users/vieting/experiments/switchboard/ctc/feat/experiments.py

```diff
@@ -776,6 +785,140 @@ def run_scf_baseline():
+        "decrease_epochs": 180,
+        "final_epochs": 0,
+    }
+
+    perturbation_args = [
+        {"speed": {"prob": 0.6, "minimum": 0.8, "maximum": 1.2}},
+        {"speed": {"prob": 0.6, "minimum": 0.7, "maximum": 1.3}},
+        {"speed": {"prob": 0.5, "minimum": 0.9, "maximum": 1.1}},
+        {"speed": {"prob": 0.5, "minimum": 0.8, "maximum": 1.2}},
+        {"speed": {"prob": 0.5, "minimum": 0.7, "maximum": 1.3}},
+        {"speed": {"prob": 0.4, "minimum": 0.9, "maximum": 1.1}},
+        {"speed": {"prob": 0.4, "minimum": 0.8, "maximum": 1.2}},
+        {"speed": {"prob": 0.4, "minimum": 0.7, "maximum": 1.3}},
+        {"tempo": {"prob": 0.4, "minimum": 0.9, "maximum": 1.1}},
+        {"tempo": {"prob": 0.4, "minimum": 0.8, "maximum": 1.2}},
+        {"tempo": {"prob": 0.4, "minimum": 0.7, "maximum": 1.3}},
+        {"tempo": {"prob": 0.5, "minimum": 0.9, "maximum": 1.1}},
+        {"tempo": {"prob": 0.5, "minimum": 0.8, "maximum": 1.2}},
+        {"tempo": {"prob": 0.5, "minimum": 0.7, "maximum": 1.3}},
+        {"tempo": {"prob": 0.6, "minimum": 0.9, "maximum": 1.1}},
+        {"tempo": {"prob": 0.6, "minimum": 0.8, "maximum": 1.2}},
+        {"tempo": {"prob": 0.6, "minimum": 0.7, "maximum": 1.3}},
+        {"preemphasis": {"prob": 0.6, "minimum": 0.9, "maximum": 1.0}},
+        {"preemphasis": {"prob": 0.6, "minimum": 0.8, "maximum": 1.0}},
+        {"preemphasis": {"prob": 0.5, "minimum": 0.9, "maximum": 1.0}},
+        {"preemphasis": {"prob": 0.5, "minimum": 0.8, "maximum": 1.0}},
+        {"codecs": [{"encoding": "ULAW", "prob": 0.4}]},
+        {"codecs": [{"encoding": "ULAW", "prob": 0.6}]},
+        {"non_linearity": {"prob": 0.4, "minimum": 0.1, "maximum": 0.3}},
+        {"non_linearity": {"prob": 0.4, "minimum": 0.2, "maximum": 0.4}},
+        {"non_linearity": {"prob": 0.6, "minimum": 0.1, "maximum": 0.3}},
+        {"non_linearity": {"prob": 0.6, "minimum": 0.2, "maximum": 0.4}},
+    ]
+
+    def process_args(args: Dict[str, Any]):
+        """
+        Process the argument dictionary to generate a key string and a report string.
+
+        Returns:
+            tuple: A tuple containing the key string and the report dict.
+        """
+
+        key_string = ""
+        report_values = {}
+
+        for key, value in args.items():
+
+            if key in ["speed", "tempo", "preemphasis", "non_linearity"]:
+                key_component = f"{key}_{value['prob']}_{value['minimum']}_{value['maximum']}"
+                key_string += key_component
+                report_values[key] = f"{value['prob']}_{value['minimum']}_{value['maximum']}"
+            elif key == "codecs":
+                codecs_str = "_".join([f"{codec['encoding']}_{codec['prob']}" for codec in value])
+                key_string += f"{key}_{codecs_str}_"
+                report_values[key] = codecs_str
+            else:
+                raise ValueError(f"Unknown argument name: {key}")
+
+        return key_string, report_dict
```

Comment:
Did you test this? You return `report_dict` here which does not seem to exist before.